### PR TITLE
Add a no-op base class for events

### LINKED ISSUE #369: Provide base class for events
Right now I'm trying to add more type information to the mitmproxy HTTP/2 stack and stumbled upon the fact that alle h2 events are just objects. I would like to see a generic `h2.events.Event` class or something similar.
Does it make sense to provide a base class for all events?

### PR DESCRIPTION
**Pull request by alexwlchan**
Fixes #369

### CODE REVIEW

**Comment by alexwlchan** [comment]
Changelog and test added.

**Comment by alexwlchan** [comment]
So the twisted test appears to be choking on the difference between

```
Usage: __main__.py [options] plugin [plugin_options]
```

and

```
Usage: trial [options] plugin [plugin_options]
```

I don’t even know where this text comes from, let alone how my change might have affected it. Before I go digging – @Lukasa, anything obvious spring to mind?

File: HISTORY.rst [hunk 1]
```diff
@@ -4,6 +4,11 @@ Release History
 2.6.0dev0
 ---------
 
+API Changes (Backward-Compatible)
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+- Added a new ``Event`` class that acts as a base class for all events.
```
Comment (Lukasa) [inline HISTORY.rst]: Let's use the fully-qualified name of this object.

File: test/test_events.py [hunk 2]
```diff
@@ -317,3 +320,26 @@ def test_alternativeserviceavailable_repr(self):
...
+        if inspect.isclass(obj):
+            yield obj
+
+
+@pytest.mark.parametrize('event', all_events())
+def test_all_events_subclass_from_event(event):
+    """
+    Every event defined in h2.events subclasses from h2.events.Event.
+    """
+    assert (event == h2.events.Event) or issubclass(event, h2.events.Event)
```
Comment (Lukasa) [inline test/test_events.py]: Let's test identity, not equality.

**Comment by alexwlchan** [comment]
Now that #372 is merged, merging current master back in. I tried to rebase, but GitHub wasn’t having it. 😕

**Comment by Lukasa** [comment]
At this point, anyone who wants to take a swing at writing some type annotations now has a decent shot of being able to fully-specify them.

**Pull request merged**